outputs/jira_test.go

```diff
@@ -0,0 +1,1074 @@
+		"name": "Time Spent",
+		"schema": map[string]interface{}{
+			"type": "number",
+		},
+	},
+	"customfield_10052": map[string]interface{}{
+		"name":   "No schema type",
+		"schema": map[string]interface{}{},
+	},
+	"customfield_10053": map[string]interface{}{
+		"name": "No schema items",
+		"schema": map[string]interface{}{
+			"type": "array",
+		},
+	},
+	"customfield_10054": map[string]interface{}{
+		"name": "Bad Type",
+		"schema": map[string]interface{}{
+			"type": "badType",
+		},
+	},
+}}
+
+func TestJiraAPI_GetName(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		jiraApi := &JiraAPI{Name: "testName"}
+		name := jiraApi.GetName()
+
+		assert.Equal(t, jiraApi.Name, name)
+	})
+}
+
+func TestJiraAPI_FetchBoardId(t *testing.T) {
+	tests := []struct {
+		name        string
+		boardName   string
+		boardList   *jira.BoardsList
+		wantJiraApi *JiraAPI
+		wantError   string
+	}{
+		{
+			name:        "happy path (0 boards found)",
+			boardName:   "board0",
+			boardList:   &jira.BoardsList{Values: []jira.Board{{Name: "board1"}, {Name: "board2"}}},
+			wantJiraApi: &JiraAPI{BoardName: "board0"},
+		},
+		{
+			name:        "happy path (1 board found)",
+			boardName:   "board1",
+			boardList:   &jira.BoardsList{Values: []jira.Board{{Name: "board1", ID: 1, Type: "Scrum"}, {Name: "board2", ID: 2, Type: "Scrum"}}},
+			wantJiraApi: &JiraAPI{boardId: 1, BoardName: "board1", boardType: "Scrum"},
+		},
+		{
+			name:        "happy path (2 boards found)",
+			boardName:   "board2",
+			boardList:   &jira.BoardsList{Values: []jira.Board{{Name: "board2", ID: 1, Type: "Scrum"}, {Name: "board2", ID: 2, Type: "Scrum"}}},
+			wantJiraApi: &JiraAPI{boardId: 2, BoardName: "board2", boardType: "Scrum"},
+		},
+		{
+			name:        "sad path (Failed to create client)",
+			boardName:   "board3",
+			wantJiraApi: &JiraAPI{BoardName: "board3"},
+			wantError:   "Failed to create client",
+		},
+		{
+			name:        "sad path (Failed to get boardList)",
+			boardName:   "board4",
+			wantJiraApi: &JiraAPI{BoardName: "board4"},
+			wantError:   "Failed to get boardList",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError == "Failed to get boardList" {
+					w.WriteHeader(http.StatusNotFound)
+				} else {
+					boardListJson, _ := json.Marshal(test.boardList)
+					_, _ = w.Write(boardListJson)
+				}
+			}))
+
+			savedCreateClient := createClient
+			createClient = func(ctx *JiraAPI) (*jira.Client, error) {
+				if test.wantError == "Failed to create client" {
+					return nil, fmt.Errorf(test.wantError)
+				} else {
+					return jira.NewClient(ts.Client(), ts.URL)
+				}
+			}
+			defer func() { createClient = savedCreateClient }()
+
+			jiraApi := &JiraAPI{BoardName: test.boardName}
+			jiraApi.fetchBoardId(test.boardName)
+
+			assert.Equal(t, test.wantJiraApi, jiraApi)
```

Comment:
don't use `test.wantJiraApi` just compare `test.boardName` and `jiraApi.BoardName`

Reply:
We should to check JiraApi, because we also [recording 'boardId' and  'boardType'](https://github.com/aquasecurity/postee/blob/127d5170b3b1624eb3ad6cdc6651cb46b2ee95fb/outputs/jira.go#L70-L76)